Habit-Tracker-JVR-Hannes/Habit-Tracker/Program.cs

```diff
@@ -0,0 +1,346 @@
+                        ClearDatabase();
+                        break;
+                    case "6":
+                        GetAllRecords(true);
+                        break;
+                    case "7":
+                        ViewHabitRecords();
+                        break;
+                    default:
+                        Console.WriteLine("\nInvalid Command. Please type a number from 0 to 7.\n");
+                        break;
+                }
+            }
+        }
+
+        private static void ViewHabitRecords()
+        {
+            Console.WriteLine("Enter the habit you want to view records for:");
+            string habitType = Console.ReadLine();
+
+            Console.Clear();
+            using (var connection = new SqliteConnection(connectionString))
+            {
+                connection.Open();
+                var tableCmd = connection.CreateCommand();
+                tableCmd.CommandText = $"SELECT * FROM habits WHERE HabitType = @habitType";
+                tableCmd.Parameters.AddWithValue("@habitType", habitType);
+
+                List<Habit> tableData = new();
+                SqliteDataReader reader = tableCmd.ExecuteReader();
+
+                if (reader.HasRows)
+                {
+                    while (reader.Read())
+                    {
+                        tableData.Add(
+                        new Habit
+                        {
+                            Id = reader.GetInt32(0),
+                            HabitType = reader.GetString(1),
+                            Date = DateTime.ParseExact(reader.GetString(2), "dd-MM-yyyy", new CultureInfo("en-US")),
+                            Quantity = reader.GetInt32(3)
+                        });
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("No records found for this habit.");
+                }
+
+                connection.Close();
+
+                Console.WriteLine("+++++++++++++++++++++++++++++++++++++++++\n");
+                foreach (var habit in tableData)
+                {
+                    Console.WriteLine($"{habit.Id} - {habit.HabitType} - {habit.Date.ToString("dd-MM-yyyy")} - Quantity: {habit.Quantity}");
+                }
+                Console.WriteLine("+++++++++++++++++++++++++++++++++++++++++\n");
+            }
+        }
+
+        private static void Update()
+        {
+            Console.Clear();
+            GetAllRecords();
+
+            var recordId = GetNumberInput("\n\nPlease type Id of the record you would like to update. Type 0 to return to main menu. \n\n");
+
+            if (recordId == 0)
+                return;
+
+            using (var connection = new SqliteConnection(connectionString))
+            {
+                connection.Open();
+
+                var checkCmd = connection.CreateCommand();
+                checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM habits WHERE Id = {recordId})";
```

Comment:
Would love to see you parameterise all of your queries like you have done elsewhere in your code.

Although, the ones I have seen are fine as you are validating user input and dealing with `int` and `DateTime`.